src/main/scala/exercises/05_Patterns.scala

```diff
@@ -56,15 +64,26 @@ object PatternMatching {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case str : String => s"A string with length ${str.length()}"
+      case n : Int => "A positive integer"
+      case p : Person => s"A person with name: ${p.name}"
+      case seq @ Seq(_ *) if seq.length > 10 => "Seq with more than 10 elements"
+      case Seq(v1, v2, tail @ _ *) => s"first: $v1, second: $v2, rest: $tail"
+      case null => "A null value"
+      case _ => "Some Scala class"
+    }
   }
 
   /**
    * If the person is older than 30, return an `Option` with the person's name;
    *    otherwise return `None`
    */
   def older(p: Person): Option[String] = {
-    error("fix me")
+    p match {
+      case Person(name, age) if age > 30 => Some(name)
```

Comment:
I should have just used an if statement here, but I had just gotten addicted to case statements from the previous problem.